core-tools.markdown

```diff
@@ -0,0 +1,345 @@
+---
+title: Learn about the core tools
+page: core-tools
+isDocumentation: true
+---
+
+# Learn about the core tools
+
+In this guide we'll take a look at a few core tools that are installed
+with the Haskell toolchain, namely, `ghc`, `runghc` and `ghci`.
+These tools can be used to compile, interpret or explore Haskell programs.
+
+Note: if you installed your Haskell toolchain using the stack tool only
+and these programs are not available, prefix the commands with
+`stack exec -- `, So `<command>` becomes `stack exec -- <command>`.
+See the documentation in the [Stack user guide](https://docs.haskellstack.org/en/stable/GUIDE/#exec) for more details.
+
+- [Compiling programs with ghc](#compiling-programs-with-ghc)
+    - [Common options for GHC](#common-options-for-ghc)
+        - [Warnings](#warnings)
+        - [Optimisations](#optimisations)
+- [An interactive environment](#an-interactive-environment)
+    - [Using external packages in ghci](#using-external-packages-in-ghci)
+- [Compiling programs - Advanced usage](#compiling-programs---advanced-usage)
+    - [Runtime options](#runtime-options)
+    - [Profiling](#profiling)
+
+First, let's start by opening your system's command line interface
+and running `ghc --version` to make sure we have successfully
+installed a Haskell toolchain:
+
+```
+➜ ghc --version
+The Glorious Glasgow Haskell Compilation System, version 9.0.1
+```
+
+If this fails, consult [the downloads page](/downloads) for information on
+how to install Haskell on your computer.
+
+## Compiling programs with ghc
+
+Running `ghc` invokes the Glasgow Haskell Compiler (GHC), and can be used to compile Haskell
+modules and programs into native executables and libraries.
+
+Create a new Haskell source file named `hello.hs`,
+and write the following code in it:
+
+```hs
+main = putStrLn "Hello, Haskell!"
+```
+
+Now, we can compile the program by invoking `ghc` with the file name:
+
+```sh
+➜ ghc hello.hs
+[1 of 1] Compiling Main             ( hello.hs, hello.o )
+Linking hello ...
+```
+
+GHC created the following files:
+
+1. `hello.hi` - Haskell interface file, we won't discuss that in this article
```

Comment:
If we're not going to talk about the hi and o files, maybe we shouldn't bring them up here? Perhaps something like "When we compile the application, GHC will dutifully creating our new executable, `hello` (or `hello.exe` on windows). You may notice a couple of other files that get generated too, but we won't worry about them in this tutorial."